File: picker/src/main/java/com/hy/picker/adapter/CateDlgAdapter.java
package com.hy.picker.adapter;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.facebook.drawee.drawable.ScalingUtils;
import com.facebook.drawee.generic.GenericDraweeHierarchy;
import com.facebook.drawee.view.SimpleDraweeView;
import com.hy.picker.PhotoContext;
import com.hy.picker.PictureSelectorActivity;
import com.hy.picker.R;
import com.hy.picker.core.util.SizeUtils;
import com.hy.picker.utils.OnItemClickListener;
import com.hy.picker.model.PhotoDirectory;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

/**
 * Created time : 2019/4/18 12:31 PM.
 *
 * @author HY
 */
public class CateDlgAdapter extends BaseAdapter {
    private final List<PhotoDirectory> mDirectories = new ArrayList<>();

    private final Drawable mDefaultDrawable;

    public CateDlgAdapter(Drawable defaultDrawable) {
        mDefaultDrawable = defaultDrawable;
    }

    public void reset(List<PhotoDirectory> directories) {
        mDirectories.clear();
        mDirectories.addAll(directories);
        notifyDataSetChanged();
    }

    public void add(int index, PhotoDirectory directory) {
        mDirectories.add(index, directory);
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return mDirectories.size();
    }

    @Override
    public PhotoDirectory getItem(int position) {
        return mDirectories.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    private int selectCateIndex = 0;

    private void selected(int index) {
        selectCateIndex = index;
        notifyDataSetChanged();
    }
    private int dp75;

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        Context context = parent.getContext();
        if (dp75==0){
            dp75= SizeUtils.px2dp(context,75);
        }

        View view = convertView;
        ViewHolder holder;
        if (convertView == null) {
            view = LayoutInflater.from(context).inflate(R.layout.picker_item_lst_catalog, parent, false);
            holder = new ViewHolder(view);
            view.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        holder.bind(position);

        return view;
    }


    private class ViewHolder {
        SimpleDraweeView image;
        TextView tvName;
        TextView tvNumber;
        ImageView selected;
        View itemView;

        private ViewHolder(View itemView) {
            this.itemView = itemView;
            image = itemView.findViewById(R.id.picker_catalog_image);
            tvName = itemView.findViewById(R.id.picker_catalog_name);
            tvNumber = itemView.findViewById(R.id.picker_catalog_photo_number);
            selected = itemView.findViewById(R.id.picker_catalog_selected);
        }

        void bind(int position) {

            boolean showSelected = selectCateIndex == position;

            PhotoDirectory item = getItem(position);

            GenericDraweeHierarchy hierarchy = image.getHierarchy();
            hierarchy.setPlaceholderImage(mDefaultDrawable, ScalingUtils.ScaleType.CENTER_CROP);
            hierarchy.setFailureImage(mDefaultDrawable, ScalingUtils.ScaleType.CENTER_CROP);
            image.setController(PictureSelectorActivity.
                    getDraweeController(image,
                            Uri.fromFile(new File(item.getCoverPath())),
                            dp75, dp75));
            tvNumber.setText(String.format(
                    PhotoContext.getContext().
                            getResources()
                            .getString(R.string.picker_picsel_catalog_number),
                    item.getPhotos().size()));

            tvName.setText(item.getName());
            selected.setVisibility(showSelected ? View.VISIBLE : View.INVISIBLE);

            itemView.setOnClickListener(v -> {
                if (position == selectCateIndex) {
//                    hideCatalog();
                    if (null!=mOnItemClickListener){
                        mOnItemClickListener.onClick(position,false);
                    }
                } else {
//                    hideCatalog();
                    if (null!=mOnItemClickListener){
                        mOnItemClickListener.onClick(position,true);
                    }
//                    selectCateIndex = position;
//                    mPicType.setText(tvName.getText().toString());
//                    MediaListHolder.currentPhotos.clear();
//                    MediaListHolder.currentPhotos.addAll(MediaListHolder.allDirectories.get(position).getPhotos());
//
//                    mCatalogAdapter.notifyDataSetChanged();
//                    mGridViewAdapter.notifyDataSetChanged();
//                    GridLayoutManager layoutManager = (GridLayoutManager) mGridView.getLayoutManager();
//                    if (layoutManager.findFirstVisibleItemPosition() != 0) {
//                        Fresco.getImagePipeline().pause();
//                        CommonUtils.postDelay(() -> mGridView.smoothScrollToPosition(0), 350);
//                    }
                    selected(position);
                }
            });
        }
    }

    private OnItemClickListener mOnItemClickListener;

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        mOnItemClickListener = onItemClickListener;
    }
}

File: picker/src/main/java/com/hy/picker/adapter/CateDlgAdapter.kt
package com.hy.picker.adapter

import android.graphics.drawable.Drawable
import android.net.Uri
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import com.facebook.drawee.drawable.ScalingUtils
import com.hy.picker.*
import com.hy.picker.model.PhotoDirectory
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.picker_item_lst_catalog.*
import java.io.File
import java.util.*

/**
 * Created time : 2019/4/18 12:31 PM.
 *
 * @author HY
 */
class CateDlgAdapter(private val defaultDrawable: Drawable) : BaseAdapter() {
    private val directories = ArrayList<PhotoDirectory>()

    private var selectCateIndex = 0
    private val dp75  by lazy{
        PhotoContext.context.dp(75f)
    }

    fun reset(directories: List<PhotoDirectory>) {
        this.directories.clear()
        this.directories.addAll(directories)
        notifyDataSetChanged()
    }

    fun add(index: Int, directory: PhotoDirectory) {
        directories.add(index, directory)
        notifyDataSetChanged()
    }

    override fun getCount() = directories.size

    override fun getItem(position: Int) = directories[position]

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    private fun selected(index: Int) {
        selectCateIndex = index
        notifyDataSetChanged()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val context = parent!!.context

        val view: View
        val holder: ViewHolder
        if (convertView == null) {
            view = View.inflate(context, R.layout.picker_item_lst_catalog, null)
            holder = ViewHolder(view)
            view.tag = holder
        } else {
            view = convertView
            holder = convertView.tag as ViewHolder
        }

        holder.bind(position)

        return view
    }


    private inner class ViewHolder
    constructor(internal val itemView: View) : LayoutContainer {
        override val containerView: View?
            get() = itemView

        init {
            pickerCatalogSelected.setColorFilter(PhotoPicker.theme.sendBgColor)
        }

        internal fun bind(position: Int) {

            val showSelected = selectCateIndex == position

            val item = getItem(position)

            if (null != item.coverPath) {
                val hierarchy = pickerCateLogImage.hierarchy
                hierarchy.setPlaceholderImage(defaultDrawable, ScalingUtils.ScaleType.CENTER_CROP)
                hierarchy.setFailureImage(defaultDrawable, ScalingUtils.ScaleType.CENTER_CROP)
                pickerCateLogImage.controller = PictureSelectorActivity.getDraweeController(pickerCateLogImage,
                        Uri.fromFile(File(item.coverPath)),
                        dp75, dp75)

            }


            pickerCatePhotoNum.text = String.format(
                    PhotoContext.context.resources
                            .getString(R.string.picker_picsel_catalog_number),
                    item.photos.size)

            pickerCateName.text = item.name
            pickerCatalogSelected.visibility = if (showSelected) View.VISIBLE else View.INVISIBLE

            itemView.setOnClickListener {
                if (position == selectCateIndex) {
                    _listener?.invoke(position, false)
                } else {
                    _listener?.invoke(position, true)
                    selected(position)
                }
            }
        }
    }


    private var _listener: ((Int, Boolean) -> Unit)? = null

    fun setOnItemClickListener(listener: (Int, Boolean) -> Unit) {
        _listener = listener
    }
}
